File: day-86/_306_Linear_Search.py
# // _306_Linear_Search.py

# // https://www.geeksforgeeks.org/problems/search-an-element-in-an-array-1587115621/1

class Solution:
    #Complete the below function
    def search(self,arr, x):
        #Your code here
        
        # APPROACH 1
        for num in range(0,len(arr)):
            if x == arr[num]:
                return num
                
        # APPROACH 2 # FAST
        # if x in arr:
        #     return arr.index(x) 
        
        return -1

# Input: 
# 1 2 3 4
# 3

# Output:
# 2

File: day-86/_306_Linear_Search.cpp
// _306_Linear_Search.cpp

// https://www.geeksforgeeks.org/problems/search-an-element-in-an-array-1587115621/1

class Solution {
public:
    // Function to search for an element in the array
    int search(vector<int>& arr, int x) {
        // APPROACH 1: Using a loop
        for (int i = 0; i < arr.size(); i++) {
            if (arr[i] == x) {
                return i;
            }
        }

        // APPROACH 2: Using std::find (alternative approach)
        // auto it = find(arr.begin(), arr.end(), x);
        // if (it != arr.end()) {
        //     return distance(arr.begin(), it);
        // }

        return -1;  // Element not found
    }
};

Input: 
1 2 3 4
3

Output:
2